src/nvim/screen.c

```diff
@@ -7027,6 +7032,26 @@ static void draw_tabline(void)
   redraw_tabline = FALSE;
 }
 
+void ui_ext_tabline_update(void)
+{
+  Array args = ARRAY_DICT_INIT;
+  ADD(args, INTEGER_OBJ(curtab->handle));
+  Array tabs = ARRAY_DICT_INIT;
+  FOR_ALL_TABS(tp) {
+    Dictionary tab_info = ARRAY_DICT_INIT;
+    PUT(tab_info, "tab", TABPAGE_OBJ(tp->handle));
```

Comment:
@justinmk  I just tested this on my remote ui. But TABPAGE_OBJ seems to break the msgpack, i.e., I can't decode the msgpack messages properly. And it's fine when I change it back to INTEGER_OBJ.